demos/defender/defender_demo_json/mqtt_operations.c

```diff
@@ -331,9 +328,9 @@ static void mqttCallback( MQTTContext_t * pMqttContext,
 static bool handlePublishResend( MQTTContext_t * pMqttContext );
 /*-----------------------------------------------------------*/
 
-static int32_t generateRandomNumber()
+static uint32_t generateRandomNumber()
 {
-    return( rand() % ( INT32_MAX ) );
+    return( rand() & UINT32_MAX );
```

Comment:
Why the `& UINT32_MAX` ?  rand() returns a int between 0 and RAND_MAX, which would have to be less than UINT32_MAX (since it is an int), right?

Reply:
Hmm, good point. Even though the docs say that `RAND_MAX` is platform dependent, you're right about the `int` type restricting to below `UINT32_MAX`.  I have removed `& UINT32_MAX`.